### PR TITLE
Letting GraphiQL use headers

### PR DESCRIPTION
**Pull request by cbjjensen**
## Hey, I just made a Pull Request!

The GraphiQL plugin uses the project found here: https://github.com/graphql/graphiql. This project allows for headers to be provided to the graph api call if a prop of headerEditorEnabled={true}  is provided to the component. Please see last comment on below thread: https://github.com/graphql/graphiql/issues/59

Example of the headers dialogue after being enabled:
image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by cbjjensen** [comment]
@freben  might you be able to assist in letting me know how to fix the CI verify step?

File: .changeset/young-sheep-impress.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@backstage/plugin-graphiql": patch
```
Comment (OrkoHunter) [inline .changeset/young-sheep-impress.md:2]: Using single quotes here `'` should fix your prettier error.

You can also do `yarn run prettier --write <file>`.

Comment (cbjjensen) [reply .changeset/young-sheep-impress.md]: Thank you!

**Review by timbonicus — APPROVED** [review]
Thanks, this seems sensible!

**Review by freben — APPROVED** [review]

File: plugins/graphiql/src/components/GraphiQLBrowser/GraphiQLBrowser.tsx [hunk 2]
```diff
@@ -73,7 +73,12 @@ export const GraphiQLBrowser = ({ endpoints }: GraphiQLBrowserProps) => {
         </Tabs>
         <Divider />
         <div className={classes.graphiQlWrapper}>
-          <GraphiQL key={tabIndex} fetcher={fetcher} storage={storage} />
+          <GraphiQL
+            headerEditorEnabled={true}
```
Comment (freben) [inline plugins/graphiql/src/components/GraphiQLBrowser/GraphiQLBrowser.tsx:77]: should remove the `={true}` here according to the linter :)

**Review by freben — APPROVED** [review]

**Pull request merged**